packages/react-magma-dom/src/components/ToggleButton/ToggleButton.tsx

```diff
@@ -0,0 +1,203 @@
+import * as React from 'react';
+import styled from '../../theme/styled';
+import { XOR } from '../../utils';
+import { css } from '@emotion/core';
+import { transparentize } from 'polished';
+
+import { ThemeContext } from '../../theme/ThemeContext';
+import { ThemeInterface } from '../../theme/magma';
+import { Button, ButtonProps, ButtonSize } from '../Button';
+import { ButtonColor } from '../Button';
+import { IconButton } from '../IconButton';
+import { IconProps } from 'react-magma-icons';
+import { ToggleButtonGroupContext } from '../ToggleButtonGroup/ToggleButtonGroup';
+
+export interface ToggleButtonTextProps extends ButtonProps {
+  /**
+   * Sets a disabled state for a button.
+   */
+  disabled?: boolean;
+  isChecked?: boolean;
+  isInverse?: boolean;
+  /**
+   * Shows the button in an active state.
+   */
+  selected?: boolean;
+  /**
+   * Changes the button size throughout the group between 'small', 'medium', and 'large'.
+   */
+  size?: ButtonSize;
+  /**
+   * @internal
+   */
+  theme?: ThemeInterface;
+}
+
+export interface ToggleButtonIconProps extends ButtonProps {
+  /**
+   * Sets the aria-label which is required for icon only buttons.
+   */
+  'aria-label'?: string;
+  /**
+   * Sets a disabled state for a button.
+   */
+  disabled?: boolean;
+  hasLabel?: boolean;
+  /**
+   * Icon which displays alongside text.
+   */
+  icon: React.ReactElement<IconProps>;
+  isChecked?: boolean;
+  isInverse?: boolean;
+  /**
+   * Shows the button in an active state.
+   */
+  selected?: boolean;
```

Comment:
`isChecked` is looking for the manually set `selected` state by `value` then lets the user remove the active toggled state like the others. `selected` is the boolean for on and off.

Reply:
Based on that, can we just have `isChecked` as the prop and `selected` continues to be the state (but is not a prop)?

Reply:
Remove `selected` from the props table. It's a variable that you need for the component logic, but the user will not be setting it. Instead, users will set `isChecked` so that does need to be a prop.